BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
```

Comment:
#### REST API Design
This POST endpoint is at the API root which does not follow good REST API design. 
The API should have a  `exchange-rate` resource and a GET endpoint to retrieve the rates. 
`GET /exchange-rate` should accept the source currency and target currencies in query strings. 

Ideally, the API design should have been done separately, and before this implementation so that resource names, endpoints, request & response contracts are defined
[Learn more about REST APIs](https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/)

Following separation of concerns principle, 
- Use a request handler instead of passing the `service.GetExchangeRates` directly.
- `IExchangeRateService` should be injected into the handler to get the exchange rates
- The handler should be responsible for extracting source currency and target currencies from the request query strings, validate both inputs, and pass inputs to the service if valid
- The handler should return success or error responses depending on the result of get rates call

We will have code similar to below
```suggestion
app.MapGet("/exchange-rates", async (
    [FromQuery] string? sourceCurrency,
    [FromQuery] string[] targetCurrencies,
    [FromServices] IExchangeRateService rateService,
    CancellationToken cancellationToken = default) =>
{
    // validate sourceCurrency and targetCurrencies
    // Use ExchangeRateService to get daily rates
    // Return success response with rates
    // OR return error response
});
```